torchaudio/csrc/ffmpeg/stream_reader/buffer/common.cpp

```diff
@@ -307,6 +350,58 @@ void write_nv12_cuda(AVFrame* pFrame, torch::Tensor& yuv) {
   // yuv[:, 1:] = uv
   yuv.index_put_({Slice(), Slice(1)}, uv);
 }
+
+void write_p010_cuda(AVFrame* pFrame, torch::Tensor& yuv) {
+  int height = static_cast<int>(yuv.size(2));
+  int width = static_cast<int>(yuv.size(3));
+
+  // Write Y plane directly
+  {
+    int16_t* tgt = yuv.data_ptr<int16_t>();
+    CUdeviceptr src = (CUdeviceptr)pFrame->data[0];
+    int linesize = pFrame->linesize[0];
+    TORCH_CHECK(
+        cudaSuccess ==
+            cudaMemcpy2D(
+                (void*)tgt,
+                width*2,
+                (const void*)src,
+                linesize,
+                width*2,
+                height,
+                cudaMemcpyDeviceToDevice),
+        "Failed to copy Y plane to Cuda tensor.");
+  }
+  // Prepare intermediate UV planes
+  torch::Tensor uv = get_buffer16({1, height / 2, width / 2, 2}, yuv.device());
+  {
+    int16_t* tgt = uv.data_ptr<int16_t>();
+    CUdeviceptr src = (CUdeviceptr)pFrame->data[1];
+    int linesize = pFrame->linesize[1];
+    TORCH_CHECK(
+        cudaSuccess ==
+            cudaMemcpy2D(
+                (void*)tgt,
+                width*2,
+                (const void*)src,
+                linesize,
+                width*2,
+                height / 2,
+                cudaMemcpyDeviceToDevice),
+        "Failed to copy UV plane to Cuda tensor.");
+  }
+  uv = uv.permute({0, 3, 1, 2});
+  using namespace torch::indexing;
+  // Write to the UV plane
+  // very simplistic upscale using indexing since interpolate doesn't support
+  // shorts
+  yuv.index_put_({Slice(), Slice(1), Slice(None, None, 2), Slice(None, None, 2)}, uv);
+  yuv.index_put_({Slice(), Slice(1), Slice(1, None, 2), Slice(None, None, 2)}, uv);
+  yuv.index_put_({Slice(), Slice(1), Slice(None, None, 2), Slice(1, None, 2)}, uv);
+  yuv.index_put_({Slice(), Slice(1), Slice(1, None, 2), Slice(1, None, 2)}, uv);
```

Comment:
I thought `Slice(1)` translate to `:1`, like in Python's `slice(1)` but the code is working so it seems to translate to `1:`. (Perhaps this is where libtorch differs from Python?)
I think `Slice(1, 3)` is better as it's more explicit that we are pushing 2 channel UV tensor.

```suggestion
  yuv.index_put_({Slice(), Slice(1, 3), Slice(None, None, 2), Slice(None, None, 2)}, uv);
  yuv.index_put_({Slice(), Slice(1, 3), Slice(1, None, 2), Slice(None, None, 2)}, uv);
  yuv.index_put_({Slice(), Slice(1, 3), Slice(None, None, 2), Slice(1, None, 2)}, uv);
  yuv.index_put_({Slice(), Slice(1, 3), Slice(1, None, 2), Slice(1, None, 2)}, uv);
```